package drizzle;

message Event {
  enum Type {
    DDL = 0;
    INSERT = 1;
    DELETE = 2;
    UPDATE = 3;
    COMMIT = 4;
  }

  message Value {
    repeated string val = 1;
  }

  required Type type = 1;
  required bool autocommit = 2;
  required string server_id = 3;
  required uint64 query_id = 4;
  required string transaction_id = 5;
  required string schema = 6;
  optional string table = 7;
  repeated string field_names = 8;
  optional string primary_key = 9;
  repeated Value values = 10;
  optional string sql = 11;
}

message EventList {
  repeated Event event = 1;
}
